sonar-scanner-engine/src/main/java/org/sonar/scanner/deprecated/perspectives/ScannerPerspectives.java

```diff
@@ -48,15 +54,27 @@ public ScannerPerspectives(PerspectiveBuilder[] builders, DefaultIndex resourceI
   public <P extends Perspective> P as(Class<P> perspectiveClass, Resource resource) {
     Resource indexedResource = resource;
     if (resource.getEffectiveKey() == null) {
-      indexedResource = resourceIndex.getResource(resource);
+      indexedResource = resourceIndex.getResource(getEffectiveKey(resource));
     }
     if (indexedResource != null) {
       PerspectiveBuilder<P> builder = builderFor(perspectiveClass);
-      return builder.loadPerspective(perspectiveClass, componentCache.get(indexedResource).inputComponent());
+      return builder.loadPerspective(perspectiveClass, componentStore.getByKey(indexedResource.getEffectiveKey()));
     }
     return null;
   }
 
+  private String getEffectiveKey(Resource r) {
+    if (r.getEffectiveKey() != null) {
+      return r.getEffectiveKey();
+    }
+
+    if (ResourceUtils.isProject(r) || /* For technical projects */ResourceUtils.isRootProject(r)) {
```

Comment:
![MINOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-minor.png 'Severity: MINOR') Remove this use of "ResourceUtils"; it is deprecated. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://next.sonarqube.com/sonarqube/coding_rules#rule_key=squid%3ACallToDeprecatedMethod)
![MINOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-minor.png 'Severity: MINOR') Remove this use of "ResourceUtils"; it is deprecated. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://next.sonarqube.com/sonarqube/coding_rules#rule_key=squid%3ACallToDeprecatedMethod)